src/main/scala/org/broadinstitute/dsde/firecloud/model/ModelSchema.scala

```diff
@@ -14,35 +13,114 @@ import org.broadinstitute.dsde.firecloud.FireCloudConfig
  * Time: 14:18
  */
 
-object ModelSchema {
+trait ModelSchema {
+  def memberTypeFromEntityType (entityType: String): Option[String]
+  def isCollectionType (entityType: String): Boolean
+  def getPlural (entityType: String): String
+  def getRequiredAttributes(entityType: String): Map[String, String]
+  def getTypeSchema(entityType: String): EntityMetadata
 
-  def getTypeSchema(entityType: String): Try[EntityMetadata] = {
-    EntityTypes.types.get(entityType) match {
-      case Some(schema) => Success(schema)
-      case None => Failure(new RuntimeException("Unknown entity type: " + entityType))
+  def isEntityTypeInSchema(entityType: String): Boolean = {
+    Try(this.memberTypeFromEntityType(entityType)) match {
+      case Failure(_) => false
+      case _ => true
+    }
+  }
+}
+
+
+
+object SchemaTypes {
+  trait SchemaType
```

Comment:
nit: should be `sealed trait`

Reply:
i wanted this trait to be extendable...
although i'm not sure that just leaving off the sealed really allows it to be extended.